### PR TITLE
simplified_pipeline_ctrl to support 2-staged pipelines

### PR DESCRIPTION
**Pull request by jmguzik**

### CODE REVIEW

File: cmd/simplified-pipeline-ctrl/reconciler.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	for _, presubmit := range presubmits.alwaysRequired {
+		if pjob, ok := latestBatch[presubmit]; ok && pjob.Status.State != v1.SuccessState {
+			return false, nil
+		}
+	}
+	for _, presubmit := range presubmits.conditionallyRequired {
+		if pjob, ok := latestBatch[presubmit]; ok && pjob.Status.State != v1.SuccessState {
+			return false, nil
+		}
+	}
```
Comment (bbguimaraes) [inline cmd/simplified-pipeline-ctrl/reconciler.go:149]: This distinction between always/conditionally required doesn't seem to be used anywhere?

Comment (bbguimaraes) [reply cmd/simplified-pipeline-ctrl/reconciler.go:149]: If we merge them, naming them something to the effect of "first stage" and "second stage" would make it easier to understand the categorization here and in the provider.

File: cmd/simplified-pipeline-ctrl/reconciler.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+	return nil
+}
+
+func composeKey(refs *v1.Refs) string {
+	return fmt.Sprintf("%d:%s/%s@%s", refs.Pulls[0].Number, refs.Org, refs.Repo, refs.BaseRef)
+}
+
+func (r *reconciler) reportSuccessOnPR(ctx context.Context, pj *v1.ProwJob, presubmits presubmitTests) (bool, error) {
```
[the remark this answers is not in the document]
Comment (bbguimaraes) [reply cmd/simplified-pipeline-ctrl/reconciler.go:106]: Or maybe make it clearer that this advances to the "second stage" of the pipeline, "success" is a bit ambiguous.

File: cmd/simplified-pipeline-ctrl/reconciler.go (hunk 1 above)
Comment (jmguzik) [reply cmd/simplified-pipeline-ctrl/reconciler.go:149]: There was a distinction but I removed it to clear logs. Thanks for the catch.

**Comment by jmguzik** [comment]
\o/

**Comment by jmguzik** [comment]
BTW this code is working on app.ci so we would see if it catches basic scenario o_0

**Pull request closed without merging**